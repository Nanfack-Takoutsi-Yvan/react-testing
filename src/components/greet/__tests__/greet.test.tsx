import { render, screen } from "@testing-library/react"
import Greet from ".."

describe("Greet snapshot", () => {
  it("should match snapshot with default hello world", () => {
    const view = render(<Greet />)

    expect(view.container).toMatchSnapshot()
  })

  it("should match snapshot with props name", () => {
    const view = render(<Greet name="Rudeus" />)

    expect(view.container).toMatchSnapshot()
  })
})

describe("Greet", () => {
  test("Greet component renders correctly", () => {
    render(<Greet />)
    const textElement = screen.getByText(/Hello/i)
    expect(textElement).toBeInTheDocument()
  })
  
  test("Greet component renders the name props correctly", () =>  {
    render(<Greet name="Rudeus" />)
    const textElement = screen.getByText("Hello Rudeus")
    expect(textElement).toBeInTheDocument()
  })
 })