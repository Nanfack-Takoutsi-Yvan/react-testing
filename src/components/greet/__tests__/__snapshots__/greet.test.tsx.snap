// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greet snapshot should match snapshot with default hello world 1`] = `
<div>
  <div>
    Hello 
  </div>
</div>
`;

exports[`Greet snapshot should match snapshot with props name 1`] = `
<div>
  <div>
    Hello 
    Rudeus
  </div>
</div>
`;
